### PR TITLE
[Site design revamp] Remove thumbnail mode selector

### PR DESCRIPTION
**Pull request by mkevins**
This PR addresses parts A, B, and C of https://github.com/wordpress-mobile/WordPress-Android/issues/16392

Primarily, this PR removes the thumbnail viewport mode selector in the site design screen. Additionally, since the changes are fairly trivial, this also removes the screen's subtitle text and changes the wording of the title.

---

**Note:** We may want to also consider updating this test:

https://github.com/wordpress-mobile/WordPress-Android/blob/d3c84c430eaaf39d5e834291be48bb08c1d31b79/WordPress/src/test/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerViewModelTest.kt#L224-L229

if we decide to default to `PreviewMode.MOBILE` even for tablets, since on the main screen we will no longer have a way to change them. Also, it should be considered whether or not we want the thumbnail preview modes to change when the user comes _back_ from the preview, since this could be confusing with no way to change the mode on that screen. :thinking: 

#### Theme (Site Design) Picker

1. Start the site creation flow
2. Choose an Intent (or skip)
3. Choose a site name (or skip)
4. Expect to be at the site design screen

Expectations:
* The title should be "Choose a theme"
* The `PreviewMode` selector should be gone
* The subtitle text should be gone

#### Page Picker

1. Navigate to the pages list
2. Tap the FAB to create a new page
3. Expect to be at the page layout picker

Expectations:
* The `PreviewMode` selector should still be present (and working as usual)
* The subtitle text should still be present

## Regression Notes
1. Potential unintended areas of impact
Modal layout picker (for page creation).

5. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested this to ensure that there are no regressions. Also, there are automated unit tests for the underlying view model.

6. What automated tests I added (or what prevented me from doing so)
The view models for the shared implementations have unit tests, but I've added a note to consider changing an existing view model test, pending further decisions on the desired UX.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt:90]: It seemed simplest to me to just remove this from the layout by setting it to be `GONE`. Alternatively, we could remove it in the xml, but since it is currently shared with the page picker, this would require forking the layouts. :man_shrugging:

File: WordPress/src/main/res/values/strings.xml
Comment (mkevins) [inline WordPress/src/main/res/values/strings.xml:3349]: This wasn't part of the task, but I guess this makes sense to change for consistency. If we want to leave it as it was, we can revert this back. Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/mlp/ModalLayoutPickerFragment.kt
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/mlp/ModalLayoutPickerFragment.kt:121]: Thank you for fixing this, no more soft-error highlighting in Android Studio 🎉

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt
Comment (ovitrif) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt:90]: It also feels to me that your approach is the more convenient one 👍

File: WordPress/src/main/res/values/strings.xml
Comment (ovitrif) [reply WordPress/src/main/res/values/strings.xml:3349]: Thank you for not missing this, I'd say we go forward with including this change in the merge of this PR 🙇

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt:89]: I think we should also remove the `hpp_subtitle` string resource. Could be actually the reason the CI failed since it's now an unused resource, didn't check yet the CI though just searched in code and I can confirm this resource is nowhere else except in `strings.xml` files.

**Review by ovitrif — APPROVED** [review]
Everything worked as expected in my tests 🎉 

Thank you @mkevins for your work, it lgtm and the only thing we may have to do before merging this is to remove the `hpp_subtitle` string resource.

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt
Comment (ovitrif) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt:89]: I opened a suggestion PR for removing the unused resource: https://github.com/wordpress-mobile/WordPress-Android/pull/16419

**Comment by twstokes** [comment]
👋 I've created some testing steps in the [WPiOS equivalent of this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/18455) that may be handy for Android.

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt:89]: I left this in because of this:

> and assess if it should become an accessibility hint

but actually, we don't have a place for that at the moment (it wouldn't go on the `TextView` itself, as those should _not_ have a `contentDescription`, though it could go on a button later in the flow. I think it's fine to remove it for now, since we can always just add it back (and it probably deserves a new name, and possibly adjusted copy too if we do reuse it). Thanks for the suggestion PR, I'll merge it here. :+1:

File: WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutListItemUiState.kt
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutListItemUiState.kt:12]: Funny how this looks like Hungarian Notation but it isn't 😅

**Review by ovitrif — APPROVED** [review]
I've tested the updates on a phone and tablet emulator with the testing steps and it works as expected.

One small issue I noticed which I'm not sure it's desired is that there is no title for the screen on tablet:

tablet

This seems to be an issue on all the new screens we added recently:
| Site Intent Question | Site Name |
| --- | --- |
| siq_tablet | sn_tablet |

Because of this I created a ticket on our internal backlog to fix the issue for every screen.

**Comment by ovitrif** [comment]
Feels ready to be merged, I'll do that after the CI checks pass. 🚀 🚀 🚀

**Comment by ovitrif** [comment]
Android Lint failed because there was another resource left unused that we missed to remove.

I pushed the change to this PR since it's trivial and it's avoidable overhead to actually create a PR just for this.

Will merge it if the checks are green this time 🤞

File: WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutListItemUiState.kt
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutListItemUiState.kt:12]: I had the same thought :laughing:

**Pull request merged**
